### ISSUE TITLE
Deck Actions

### ISSUE THREAD

**Issue by D-D-2** [DeckofAdventures/TheGame#63]
## Feature Request
Currently the game has 2 Actions. I'd like to see the possibility of having Action, Bonus Action, Deck Action. A deck of cards are only semi-random, and that randomness actively declines as the game progresses. In order to take of the inherent strategic possibilities that a deck of cards lends itself to, I suggest "Deck Actions" be added to the game. These actions do not have any direct impact on the game/RP, but as a result have many possible indirect impacts. They also allow players to employ a strategic mindset during the entirety of of an encounter, rather than only on their turn. It is now important to keep track of my discard pile for more than just counting cards and knowing what I have left. Perhaps I can even start to keep track of what cards the GM has discarded, or my fellow players. These actions add a thick layer of complexity, player agency, and engagement. 

### Current gap
Currently there is no deck interaction that I am aware of, except for the possible future additional powers that can be used on the deck/discard pile. With a deck action, it allows the player to not only interact with the game from a traditional TTRPG standpoint, but also as a strategy game with resource management. This will increase the drift from traditional dice TTRPGs and increase rationale for the question of "why use a deck of cards." 

### Suggested resolution 

Examples of Deck Actions: 

- As a deck action, reshuffle 3 random cards back into your deck from your discard pile. 
- As a deck action, place 1 card from your discard pile back in your deck, 3rd from the top 
- As a deck action, banish the top card from a target creature’s deck 
- As a deck action, swap the top card of your deck with a random deck master, face down. 
- As a deck action, banish the top 4 cards from your deck. Gain one FP point. 
- As a deck action, banish your discard pile. Gain one FP point for every 10 cards banished this way. 
- As a deck action, place the top card of your deck face down in a separate pile. As a bonus action, you may turn a card from this pile. 
   - If it is a card of your suit, gain one FP and discard this card. 
   - If it is an Ace, gain 2FP and banish this card. 
   - If it is an Ace of your suit, gain 3FP and banish this card. 
   - If it is none of these, banish the turned card.
- If your target card is of your suit, you may use a deck action to draw an additional card. If the drawn card is of your suit, deal one damage to a visible target creature within 6 squares of you. Discard this card. 
- If your target card would be an Ace, you may use a deck action to place it in your FP reserve and redraw your target card. Discard three cards from the top of your deck, and banish any aces discarded this way.

**Comment by D-D-2** [comment]
Yes, I don't see there being too many deck actions making it into the final game. I think we can take the "less is more" philosophy on this one. Something to consider is will certain deck actions be archetype specific? I think it makes thematic sense for some, but not sure

**Comment by LockerM** [comment]
Thought from @CBroz1 - Could we leverage our 'Discard' actions and the proposed Deck Actions to replace 'Minor' Actions for certain characters who take specific Powers?

**Comment by LockerM** [comment]
Can we make a set of Powers 'archetype agnostic' and just level-require them to allow people to take more Powers that modify the set of "Deck Actions" that are used as a Minor Action. 

Making a subset of Powers that allow players to choose from a set of Deck Actions, or 'attune' to only 2 Deck Actions so people can find synergies between the Deck Actions and their archetype Powers.

**Comment by LockerM** [comment]
Proposal from conversation - A "Deck Action" is a sub-category of Powers that are specifically labeled as Deck Actions? Or "Game Mechanics" or "Deck Mechanics" as a tag. 

These Powers won't allow for additional damage, etc., but mostly affect the Deck and interact with these Powers themselves.

**Comment by LockerM** [comment]
Next steps: Find a core set of 2-3 "Deck Actions" as Minor Powers to add to the Powers List to have a good sense of the shape of what to build next.
